internals/daemon/api_checks.go

```diff
@@ -114,6 +114,48 @@ func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
 	return SyncResponse(responsePayload{Changed: changed})
 }
 
+func v1PostChecksRefresh(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Name string `json:"name"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+
+	if payload.Name == "" {
+		return BadRequest("must specify check name")
+	}
+
+	plan := c.d.overlord.PlanManager().Plan()
+	check, ok := plan.Checks[payload.Name]
+	if !ok {
+		return NotFound("cannot find check with name %q", payload.Name)
+	}
+
+	checkMgr := c.d.overlord.CheckManager()
+	result, err := checkMgr.RefreshCheck(r.Context(), check)
+	info := checkInfo{
+		Name:      result.Name,
+		Level:     string(result.Level),
+		Startup:   string(result.Startup),
+		Status:    string(result.Status),
+		Failures:  result.Failures,
+		Threshold: result.Threshold,
+		ChangeID:  result.ChangeID,
+	}
+	errStr := ""
+	if err != nil {
+		errStr = err.Error()
+	}
+
+	type responsePayload struct {
```

Comment:
It's a bit annoying that there's a top-level type called `responsePayload` (that was probably me; we could make that one defined locally too). In any case, probably just `payload` is enough here. Could do `refreshPayload`, but I think the `refresh` bit wouldn't be needed in this context.

Reply:
There already is a `payload` (request payload), so I'm gonna use `refreshPayload`.